File: nvxapp.client/src/nvx/ClientServer-Service/GestionePresenze/Par_Competenza/Models/par-competenza-model.ts
import { ModelResult } from "../../../ModelsBase/model-result";

export class Par_CompetenzaModel {
  public id!: number;
  public idAz_Anagrafica!: number;
  public descrizione: string;
  public default: boolean;
}

export class Par_Competenza_GetAll_InModel {}

export class Par_Competenza_GetAll_OutModel extends ModelResult {
  public par_Competenza: Par_CompetenzaModel[] = [];
}


export class Par_CompetenzaGetInModel {
  public id: number = 0;
}

export class Par_CompetenzaGetOutModel extends ModelResult {
  public par_Competenza: Par_CompetenzaModel = new Par_CompetenzaModel();
}


export class Par_CompetenzaPutInModel extends ModelResult {
  public par_Competenza: Par_CompetenzaModel = new Par_CompetenzaModel();
}

export class Par_CompetenzaPutOutModel extends ModelResult {
  public par_Competenza: Par_CompetenzaModel = new Par_CompetenzaModel();
}


export class Par_CompetenzaDeleteInModel {
  public id: number = 0;
}

export class Par_CompetenzaDeleteOutModel extends ModelResult {
  public par_Competenza: Par_CompetenzaModel = new Par_CompetenzaModel();
}

File: nvxapp.server.service/ClientServer-Service/GestionePresenze/Par_CompetenzaService/Models/Par_CompetenzaModel.cs
using nvxapp.server.service.ClientServer_Service.ModelsBase;
using System.Collections.Generic;

namespace nvxapp.server.service.ClientServer_Service.GestionePresenze.Par_CompetenzaService.Models
{
    public class Par_CompetenzaModel
    {
        public  int Id { get; set; }
        public  int IdAz_Anagrafica { get; set; }
        public string Descrizione { get; set; }  = string.Empty;
    }

    public class Par_Competenza_GetAll_InModel
    {
    }

    public class Par_Competenza_GetAll_OutModel : ModelResult
    {
        public List<Par_CompetenzaModel> Par_Competenza { get; set; } = new List<Par_CompetenzaModel>();
        public Par_Competenza_GetAll_OutModel() { }
    }

    public class Par_CompetenzaGetInModel
    {
        public int Id { get; set; } = 0;
    }

    public class Par_CompetenzaGetOutModel : ModelResult
    {
        public Par_CompetenzaModel Par_Competenza { get; set; } = new Par_CompetenzaModel();
    }

    public class Par_CompetenzaPutInModel : ModelResult
    {
        public Par_CompetenzaModel Par_Competenza { get; set; } = new Par_CompetenzaModel();
    }

    public class Par_CompetenzaPutOutModel : ModelResult
    {
        public Par_CompetenzaModel Par_Competenza { get; set; } = new Par_CompetenzaModel();
    }
}
